pkg/ruler/scheduler.go

```diff
@@ -182,6 +184,18 @@ func (s *scheduler) poll() (map[string]configs.VersionedRulesConfig, error) {
 func (s *scheduler) addNewConfigs(now time.Time, cfgs map[string]configs.VersionedRulesConfig) {
 	// TODO: instrument how many configs we have, both valid & invalid.
 	level.Debug(util.Logger).Log("msg", "adding configurations", "num_configs", len(cfgs))
+	cycleNanos := float64(s.evaluationInterval.Nanoseconds())
+	nextEvalCycle := time.Unix(0, int64(math.Ceil(float64(now.UnixNano())/cycleNanos)*cycleNanos))
+	hasher := fnv.New64a()
```

Comment:
What's up with constructing this out here, rather than inside the function? I'm guessing it's an optimization, but if so, why are we constructing it per `addNewConfigs` invocation rather than once per process?

Reply:
Yes, it's at that level as an optimisation. The hashing object is stateful; I wanted to keep it as local as possible to avoid the need to add mutexes.